core/src/Platforms/net45/NetDesktopPlatformProxy.cs

```diff
@@ -204,6 +207,22 @@ public string GetProductName()
             return _isMsal ? "MSAL.Desktop" : "PCL.Desktop";
         }
 
+        public string GetApplicationName()
+        {
+            return Assembly.GetEntryAssembly().GetName().Name.ToString();
+        }
+
+        public string GetApplicationVersion()
+        {
+            return Assembly.GetEntryAssembly().GetName().Version.ToString();
+        }
+
+        public string GetDeviceId()
+        {
+            return NetworkInterface.GetAllNetworkInterfaces().Where(nic => nic.OperationalStatus == OperationalStatus.Up)
+                .Select(nic => nic.GetPhysicalAddress().ToString()).FirstOrDefault();
```

Comment:
Isn't this PII ?

Reply:
Device Id is PII. It's being hashed before it's set in DefaultEvent. We are doing all hashing in the telemetry classes, so I put it there for consistency. I can bring it here if that makes more sense though.

Reply:
I think you could make a comment in the code here, so it's clear the hashing is happening elsewhere.

---
In reply to: [228231893](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1352#discussion_r228231893) [](ancestors = 228231893)

Reply:
also, where exactly is that other place?

---
In reply to: [228346528](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1352#discussion_r228346528) [](ancestors = 228346528,228231893)

Reply:
agree there should be a fallback

---
In reply to: [228073704](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1352#discussion_r228073704) [](ancestors = 228073704)

Reply:
Hashed in DefaultEvent